core/src/test/java/org/elasticsearch/indexing/IndexActionIT.java

```diff
@@ -223,4 +229,35 @@ public void testInvalidIndexName() {
                     e.getMessage().contains("Invalid index name [..], must not be \'.\' or '..'"), equalTo(true));
         }
     }
+
+    /**
+     * This test ensures that indexing operations adhere to the {@link IndexSettings#WAIT_FOR_ACTIVE_SHARDS_SETTING}.
+     */
+    public void testChangeWaitForActiveShardsSetting() throws Exception {
```

Comment:
We have these tests in `ReplicationOperationTests`, so I will add them for `TransportReplicationActionTests` and get rid of it here.